### PR TITLE
Test case and workaround for Issue 1696. Fixes #1696.

### LINKED ISSUE #1696: crash in ElementAnnotationUtil.getTypeAtLocation
```java
import org.checkerframework.checker.nullness.qual.Nullable;

interface T {
  interface I<T extends I<T, S>, S> {}
  void f(@Nullable I<?, ?> x);
}
```

```
./checker-framework-2.3.0/checker/bin/javac -cp checker-framework-2.3.0/checker/dist/checker-qual.jar -processor Nullness -AprintErrorStack T.java 
...
error: ElementAnnotationUtil.getTypeAtLocation: only declared types, arrays, and null types can have annotations with location; found type: capture#7 of ? extends Object (kind: TYPEVAR) location: [WILDCARD]
  Compilation unit: T.java
  Exception: java.lang.Throwable; Stack trace: org.checkerframework.framework.source.SourceChecker.errorAbort(SourceChecker.java:726)
  org.checkerframework.javacutil.ErrorReporter.errorAbort(ErrorReporter.java:25)
  org.checkerframework.framework.util.element.ElementAnnotationUtil.getTypeAtLocation(ElementAnnotationUtil.java:384)
  org.checkerframework.framework.util.element.ElementAnnotationUtil.getLocationTypeADT(ElementAnnotationUtil.java:429)
  org.checkerframework.framework.util.element.ElementAnnotationUtil.getTypeAtLocation(ElementAnnotationUtil.java:374)
  org.checkerframework.framework.util.element.ElementAnnotationUtil.getLocationTypeAWT(ElementAnnotationUtil.java:481)
  org.checkerframework.framework.util.element.ElementAnnotationUtil.getTypeAtLocation(ElementAnnotationUtil.java:376)
  org.checkerframework.framework.util.element.ElementAnnotationUtil.getLocationTypeADT(ElementAnnotationUtil.java:429)
  org.checkerframework.framework.util.element.ElementAnnotationUtil.getTypeAtLocation(ElementAnnotationUtil.java:374)
  org.checkerframework.framework.util.element.ElementAnnotationUtil.annotateViaTypeAnnoPosition(ElementAnnotationUtil.java:279)
  org.checkerframework.framework.util.element.ParamApplier.handleTargeted(ParamApplier.java:213)
  org.checkerframework.framework.util.element.TargetedElementAnnotationApplier.extractAndApply(TargetedElementAnnotationApplier.java:193)
  org.checkerframework.framework.util.element.ParamApplier.extractAndApply(ParamApplier.java:251)
  org.checkerframework.framework.util.element.ParamApplier.apply(ParamApplier.java:49)
  org.checkerframework.framework.type.ElementAnnotationApplier.apply(ElementAnnotationApplier.java:94)
  org.checkerframework.framework.type.TypeFromMemberVisitor.visitVariable(TypeFromMemberVisitor.java:32)
  org.checkerframework.framework.type.TypeFromMemberVisitor.visitVariable(TypeFromMemberVisitor.java:21)
  com.sun.tools.javac.tree.JCTree$JCVariableDecl.accept(JCTree.java:864)
  com.sun.source.util.SimpleTreeVisitor.visit(SimpleTreeVisitor.java:53)
```

### PR DESCRIPTION
**Pull request by wmdietl**

### CODE REVIEW

File: framework/src/org/checkerframework/framework/util/element/ElementAnnotationUtil.java [hunk 1]
```diff
@@ -369,29 +370,40 @@ static AnnotatedTypeMirror getTypeAtLocation(
...
         }
+        switch (type.getKind()) {
+            case NULL:
+                return getLocationTypeANT((AnnotatedNullType) type, location);
+            case DECLARED:
+                return getLocationTypeADT((AnnotatedDeclaredType) type, location);
+            case WILDCARD:
+                return getLocationTypeAWT((AnnotatedWildcardType) type, location);
+            case TYPEVAR:
+                if (TypesUtils.isCaptured((TypeVariable) type.getUnderlyingType())) {
```
Comment (wmdietl) [inline framework/src/org/checkerframework/framework/util/element/ElementAnnotationUtil.java]: This is the work-around for Issue #1696. The other changes in this PR remove single-use static imports which made the code harder to read.

**Comment by wmdietl** [comment]
Let's not squash this PR.

**Comment by mernst** [comment]
Could you please separate this into multiple pull requests, one for each distinct issue?

**Comment by wmdietl** [comment]
Now this is only #1696

**Pull request merged**